corehq/apps/userreports/views.py

```diff
@@ -138,6 +139,7 @@ def create_report(request, domain):
 
 class ReportBuilderView(BaseDomainView):
 
+    @method_decorator(require_permission(Permissions.edit_data))
     @cls_to_view_login_and_domain
```

Comment:
I am not quite sure I understand your question. This uses the existing `Permissions.edit_data` to restrict access to Report builder to web users that have the edit data permission